### PR TITLE
chore(ci): Migrate phone number environments

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

This migrates phone environments out of the main AWS account. 
- Updated `environments/` to point to the new AWS resources in e2e AWS account.
- Updated `environments/auth/package.json` to simplify commands
- Updated workflows to reflect these changes

#### Important Note
PR checks against this PR **will fail**, because this PR updates workflows and they won't take affect they're merged. I've done separate POC workflows, listed below. I will force merge this PR once approved.

#### Description of how you validated changes

POC on #3362.

- [POC / Build and Runtime Test](https://github.com/aws-amplify/amplify-ui/actions/workflows/poc-canary-test.yml)
- [POC / Test / PRs](https://github.com/aws-amplify/amplify-ui/actions/workflows/poc-test-prs.yml)
- [POC / Test and Publish / next](https://github.com/aws-amplify/amplify-ui/actions/workflows/poc-publish-next.yml)
- [POC / Test and Publish / latest](https://github.com/aws-amplify/amplify-ui/actions/workflows/poc-publish-latest.yml)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: environments/auth/package.json [hunk 1]
```diff
@@ -3,9 +3,7 @@
   "name": "environments-auth",
   "version": "0.0.1",
   "scripts": {
-    "pull": "../pull-environments.sh",
-    "pull:phone": "yarn pull -r us-east-1 -i \"\\./auth-with-((phone-and-sms-mfa)|(phone-number)|(totp-and-sms-mfa))\"",
-    "pull:email": "yarn pull -r us-east-2 -i \"\\./auth-with-email\"",
-    "pull:non-phone": "yarn pull -r us-east-2 -e \"\\./auth-with-((phone-and-sms-mfa)|(phone-number)|(totp-and-sms-mfa))\""
+    "pull": "../pull-environments.sh -r us-east-2",
+    "pull:email": "yarn pull -i \"\\./auth-with-email\""
```
Comment (wlee221) [inline environments/auth/package.json:7]: Leaving `pull:email` as-is because it's used specifically by guides test.

**Review by ioanabrooks — APPROVED** [review]
LGTM

File: .github/workflow-samples/tagged-release.yml [hunk 2]
```diff
@@ -24,8 +24,6 @@ jobs:
       commit: ${{ github.sha }}
       repository: ${{ github.repository }}
     secrets:
-      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }} # TODO: Remove this
-      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }} #TODO: Remove this
```
Comment (wlee221) [inline .github/workflow-samples/tagged-release.yml:28]: ✨

Comment (reesscot) [reply .github/workflow-samples/tagged-release.yml:28]: 💯

**Review by reesscot — APPROVED** [review]
Thanks for fixing this, this is a big ops win!

**Pull request merged**